WordPress/gutenberg
# Add initial block support for width

## Description

Issue #26368: This adds a dimensions block support providing a width selector with default options for 25%, 50%, 75%, and 100% width.

This is an alternative to implementing width as a selector directly on the Button block, implemented here: (#25999)

### Changes
* Adds a new dimensions block support
* Provides default width options using CSS variables, allowing for preset options through theme customization
* Updates the `Button` component to use the new block support

### Notes
* The `width` options are nested under `dimensions` settings and could be expanded to include additional block support options for height
* This is a first iteration and could be iterated on to include custom percentage widths as a text entry, or px width support similar to what's implemented in the `search` block
* The support for `Button` sets the width of the entire component -- no adjustments were made for margins in this iteration, so for example you will find that a 25% width button does not fit on the same line as a 75% width button.


## How has this been tested?

Manual testing using the Button block.

**Test instructions**

1. Add a `Buttons` block to a post, with one or more `Button`s.
2. Select a `Button`
3. You should see a Width Settings section under the Inspector Controls, with options for 25%, 50%, 75%, 100%.
4. Toggle a width option and **verify that the selected button updates to the correct width. You should see an inline-style added to the block using `var(--wp--preset--width--<the width you selected>)`.**
5. Save the post and view on the frontend.
6. **The same inline styling should be applied in the saved post.**


## Screenshots 

![width-block-support](https://user-images.githubusercontent.com/63313398/96180695-ab5a7b00-0ee7-11eb-8997-7e8b4e0d789b.gif)

## Types of changes
New feature (non-breaking change which adds functionality)

## Next Steps


## Checklist:
